src/domain/service_domain_db.erl

```diff
@@ -0,0 +1,141 @@
+-module(service_domain_db).
+
+-behaviour(mongoose_service).
+
+-include("mongoose_config_spec.hrl").
+-include("mongoose_logger.hrl").
+
+-define(GROUP, service_domain_db_group).
+
+-export([start/1, stop/0, config_spec/0]).
+-export([start_link/0]).
+-export([enabled/0]).
+-export([force_check_for_updates/0]).
+-export([sync/0]).
+
+%% gen_server callbacks
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2,
+         terminate/2, code_change/3]).
+
+%% ---------------------------------------------------------------------------
+%% Client code
+
+start(Opts) ->
+    mongoose_domain_sql:start(Opts),
+    mongoose_domain_db_cleaner:start(Opts),
+    ChildSpec =
+        {?MODULE,
+         {?MODULE, start_link, []},
+         permanent, infinity, worker, [?MODULE]},
+    supervisor:start_child(ejabberd_sup, ChildSpec),
+    ok.
+
+stop() ->
+    mongoose_domain_db_cleaner:stop(),
+    supervisor:terminate_child(ejabberd_sup, ?MODULE),
+    supervisor:delete_child(ejabberd_sup, ?MODULE),
+    ok.
+
+-spec config_spec() -> mongoose_config_spec:config_section().
+config_spec() ->
+    #section{items = #{
+               <<"event_cleaning_interval">> => #option{type = integer,
+                                                        validate = positive},
+               <<"event_max_age">> => #option{type = integer,
+                                              validate = positive}
+              }}.
+
+start_link() ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
+
+enabled() ->
+    is_pid(whereis(?MODULE)).
```

Comment:
This check seems prone to race conditions, why not just check if the service is enabled in the config?